lib/rucio/web/rest/flaskapi/v1/dids.py

```diff
@@ -654,8 +661,13 @@ def get(self, scope_name):
         """
         try:
             scope, name = parse_scope_name(scope_name, request.environ.get('vo'))
-            dynamic = 'dynamic' in request.args
-            did = get_did(scope=scope, name=name, dynamic=dynamic, vo=request.environ.get('vo'))
+            dynamic_depth = None
+            if 'dynamic_depth' in request.args:
+                orig = request.args['dynamic_depth'].upper()
+                dynamic_depth = DIDType.DATASET if orig == 'DATASET' else DIDType.FILE
```

Comment:
This has the side effect that every value gets translated to a `DIDType.File`. (e.g. `request.args['dynamic_depth'] = "Bratwurst"`). Do we want that?

```text
If set to None, or to a value which doesn't make sense (ex: requesting depth = CONTAINER for a did of type DATASET) will not compute the size dynamically.
```